core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/DeleteExpiredMetrics.java

```diff
@@ -108,7 +111,10 @@ public Completable call(JobDetails jobDetails) {
         }
 
         return expirationIndexResults
-                .flatMap(metricId -> metricsService.deleteMetric(metricId))
+                .concatMap(metricId -> metricsService.deleteMetric(metricId))
+                .doOnError(e -> {
```

Comment:
This logs the error which is good, but we still terminate the stream early. You need ```onErrorResumeNext``` or something similar.